### PR TITLE
Update builders to Xcode 13 and macOS 12

### PR DESCRIPTION
**Pull request by jmagman**
Fixes https://github.com/flutter/flutter/issues/99698
Fixes https://github.com/flutter/flutter/issues/85555

Blocking https://github.com/flutter/flutter/issues/94078

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: .ci.yaml
Comment (jmagman) [inline .ci.yaml:64]: Something like this
https://flutter.googlesource.com/recipes/+/refs/heads/main/recipe_modules/osx_sdk/api.py#110

File: .ci.yaml
Comment (jmagman) [inline .ci.yaml:78]: https://flutter-review.googlesource.com/c/infra/+/27742/comments/6a769d18_cc8d83ec

File: .ci.yaml
Comment (jmagman) [inline .ci.yaml:60]: I'm not sure what `old_osx_sdk` should be renamed to given https://github.com/flutter/flutter/pull/99586 since the names were different between the engine and framework.

File: .ci.yaml
Comment (jmagman) [inline .ci.yaml:60]: I'm not sure what `old_osx_sdk` should be renamed to given https://github.com/flutter/flutter/pull/99586 since the names were different between the engine and framework.  `osx_sdk_13a233_13_15` is for Xcode 13a233 + iOS 13 + iOS 15 but I'm open to suggestions here.

File: .ci.yaml
Comment (jmagman) [inline .ci.yaml:243]: The web tests should run on PRs like this one that change their configurations.

**Comment by jmagman** [comment]
Tests will fail until https://github.com/flutter/cocoon/pull/1652 is published.

File: .ci.yaml
Comment (yusuf-goog) [inline .ci.yaml:76]: Some places, these are specified as ios-13-0. Not sure if it matters?

**Review by yusuf-goog — APPROVED** [review]
LGTM with a question. @godofredoc should Ok the cache related changes.

File: .ci.yaml
Comment (godofredoc) [reply .ci.yaml:76]: In the past we were tracking minor versions just to be sure that updates were not breaking the builds and or generating different performance results.

**Review by godofredoc — APPROVED** [review]

**Comment by jmagman** [comment]
Both of these are running with old versions of macOS and Xcode? The rest of the tests picked up the properties though.
https://ci.chromium.org/p/flutter/builders/try/Mac%20Unopt/9950?
https://ci.chromium.org/p/flutter/builders/try/Mac%20Web%20Engine/23525?

**Comment by jmagman** [comment]
Rebasing to run them again on published https://github.com/flutter/cocoon/pull/1652

File: .ci.yaml
Comment (jmagman) [reply .ci.yaml:76]: You were right, these need to match the cipd refs, `ios-13-0` etc
```
#0 mac_toolchain/install.go:311 - main.resolveRuntimeRef()
  reason: Failed to resolve runtime ref given runtime version: ios-13, xcode version: .
```
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8820126450450034161/+/u/install_xcode_runtime_ios-13/stdout

**Comment by CaseyHillers** [comment]
> Rebasing to run them again on published https://github.com/flutter/cocoon/pull/1652[](https://github.com/jmagman)

Sorry to be nitty, but for future infra issues can you merge instead of rebase? GitHub wipe's check run state on rebase so it's hard for me to take action on why the properties weren't propagating

**Review by yusuf-goog — APPROVED** [review]

**Comment by jmagman** [comment]
> > Rebasing to run them again on published [https://github.com/flutter/cocoon/pull/1652[](https://github.com/jmagman)](https://github.com/flutter/cocoon/pull/1652%5B%5D(https://github.com/jmagman))
> 
> Sorry to be nitty, but for future infra issues can you merge instead of rebase? GitHub wipe's check run state on rebase so it's hard for me to take action on why the properties weren't propagating

Will do.  🙂

**Comment by jmagman** [comment]
`Mac Web Engine` `felt ios-safari test` tests timing out, filed https://github.com/flutter/flutter/issues/99885

**Comment by chinmaygarde** [comment]
@jmagman I believe a rebase ought to unblock this patch. Are there blockers other than presubmits here?

**Comment by jmagman** [comment]
This is blocked on https://github.com/flutter/flutter/issues/99885.  Making a draft.

File: .ci.yaml
Comment (godofredoc) [reply .ci.yaml:60]: old_osx_sdk and osx_sdk were used to transition between two versions of xcode without impacting the builders. Now that we need multiple caches updating the name with version string is a better option.

**Review by godofredoc — APPROVED** [review]
Thanks Jenn!

**Comment by jmagman** [comment]
`Mac Web Engine` [passed in draft](https://ci.chromium.org/p/flutter/builders/try/Mac%20Web%20Engine/23967) but now starting the iOS simulator Safari browser[ timing out during `Run tests on iOS Safari`](https://ci.chromium.org/p/flutter/builders/try/Mac%20Web%20Engine/23994?), it isn't even getting to the `felt ios-safari test`.

**Comment by jmagman** [comment]
Rebasing to kick all the tests again.

**Review by godofredoc — APPROVED** [review]

**Comment by CaseyHillers** [comment]
The ci.yaml roller failed on this PR - https://ci.chromium.org/ui/p/flutter/builders/staging/Linux%20ci_yaml%20engine%20roller/2084/overview

The caches change contains invalid characters. You'll need to swap the special characters to only be underscores

**Pull request merged**
